### DISCUSSION TITLE
Analysis of grouping NFTs by collections

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#223] [General]
As you may already know, our alpha NFT functionality brought a temporary solution for grouping NFTs in an account based on the NFT creator address. Such a group is "named" by truncating the middle part of the address and replacing it with `..` so our users see a bunch of D5J7H..W44CA, FROGJ..HRW2M, and similar values.

NFT collections, respectively their names are the planned way for grouping and presenting NFTs for our users, so here comes an analysis of the data we have collected regarding NFT collections in the Algorand ecosystem. 

## Analysis

We have retrieved at least one collection name for 737,114 NFTs.

### 1) Single collection name assigned to an asset

599,137 of those or 81.3% have got a single collection name.

### 2) Multiple collection names assigned

The rest of the 137,977 NFTs are listed below:

|Names count  | NFTs count|
| :---:  | :---:              |
| 2  | 126,851             |
| 3  | 8,053            |
| 4  | 1,381             |
| 5  | 659             |
| 6  | 698             |
| 8  | 169            |
| 9  | 166             |

We analyzed similarities between those and here are the results (1.0 means equal, while 0.0 means no similarity at all):

|Similarity  | NFTs count|
| :---:  | :---:              |
| >0.76  | 69,832             |
| <0.76  | 8,502            |
| <0.67  | 28,496            |
| <0.51  | 5,254             |
| <0.34  | 20,034             |

For two names we just compare them, while for 3 and more we produced two that are most similar in collection and then we use those two.

### 3) No collection names retrieved

There are around 100,000 NFTs (the exact number will be edited here in the following hours) we don't have any collection name assigned. 

## Outcome

We should discuss whether we are ready or not to start implementing NFT collection names in our engine.

For the assets with multiple collection names, one of the solutions would be to pick a smaller size name for the actual name if the similarity is above 0.67 or 0.5. The rest of the assets will get the *old* name where we just use a truncated creator address to group by. That old way should also be used for those assets we don't have any information about their collections.

**Comment by moloch10** [comment]
I'm definitely in favor of implementing collection names. With %80+ having a single name, I can't see why that shouldn't go live.
Regarding the others, realistically, no matter how well they are matched, there will be false positives. Your proposed solution sounds reasonable and I'm sure once feedback comes in, you'll be able to tweak it and make it even more precise.  😀

**Comment by ipaleka** [comment]
We'd need analysis and/or discussion for a few gargantuan collections we're aware of, like FIFA+Collect or NFD.

### FIFA+Collect

Their names are kind of cryptic, like [D01-20 #7654](https://algoexplorer.io/asset/880915303) or [D01-39 #45077](https://algoexplorer.io/asset/889634347).

The first and obvious choice is to use that base name (D01-20, D01-39, ...) and add FIFA as a prefix, but we'd surely benefit from some more insights...

### NFD

A solution here is kind of straightforward as NFD provides us a system here: https://docs.nf.domains/fundamentals/pricing.

So, by checking the number of letters in the name and the entry price, we can get 19 collections in total.

### Other

Please add up...
